packages/yarnpkg-libzip/artifacts/build.sh

```diff
@@ -103,9 +103,35 @@ LIBZIP_REPO=arcanis/libzip
         ./libzip-"$LIBZIP_VERSION"/build/local/lib/libzip.a \
         ./zlib-"$ZLIB_VERSION"/build/local/lib/libz.a
 
+    cat > ../sources/libzip.node.js \
+        <(echo "var frozenFs = Object.assign({}, require('fs'));") \
+        <(sed 's/require("fs")/frozenFs/g' ./build.js | sed 's/process\["on"\]/(function(){})/g') \
+
+    echo Built wasm node sync version
+)
+
+(
+    source "$EMSDK"
+
+    emcc \
+        -o ./build.js \
+        -s WASM=1 \
+        -s EXPORTED_FUNCTIONS="$(cat ./exported.json)" \
+        -s EXTRA_EXPORTED_RUNTIME_METHODS='["cwrap", "getValue"]' \
+        -s ALLOW_MEMORY_GROWTH=1 \
+        -s BINARYEN_ASYNC_COMPILATION=0 \
```

Comment:
Hm I think this is the flag that should be changed, rather than the `ENVIRONMENT=node`. I think we will want to use the Node API in any case, as otherwise I have no idea what filesystem will Emscripten try to feed the libzip 🤔

Toggling `BINARYEN_ASYNC_COMPILATION` back on, however, will allow us to compile the Wasm chunk outside of the main thread, which is a requirement for it to work.

Also, can you turn the block into a shell function parametrized on `BINARYEN_ASYNC_COMPILATION` and the build name? This way we have way less risks to introduce deltas in the flag used etc.